modules/geo-layers/src/mvt-layer/mvt-layer.ts

```diff
@@ -303,7 +305,7 @@ export default class MVTLayer<
     };
   }
 
-  private getHighlightedObjectIndex(tile: Tile2DHeader): number {
+  private getHighlightedObjectIndex(tile: Tile2DHeader<MVTParsedTile>): number {
```

Comment:
I am not sure if we should add yet another dimension for TileLayer, but here's how it would like.

However, this piece of code shows that we have proper typings almost everywhere and perfectly understands current code knowing that here we're operating  on `Feature[] | BinaryFeatures`

Reply:
One concern is that when I traverse a group of TIleHeaders in general code that just accesses the common fields, I may not know what type they were created with. This parametrization can thus become quite "contagious" if I need to parametrize any code that touches headers...